taxjar/taxjar-magento2-extension
# INT-1336 Backup rate deletion

### Context

The original issue describes the need to asynchronously delete backup rates when TJ Backup Rates are disabled.

### Description

The functionality described in issue was already added in PR #195. This branch explicitly casts some retrieved config values from string > int > bool.

### Performance

No change

### Testing

1. Enable TaxJar Magento 2 extension and sync one or more nexus regions from TaxJar admin account.
3. From two separate terminal processes, run CLI commands `bin/magento queue:consumers:start taxjar.backup_rates.create` and `bin/magento queue:consumers:start taxjar.backup_rates.delete`
4. Login to backend as admin and navigate to `Stores > Configuration > Sales > Tax`
5. Enable "Backup rates"
6. Observe that "Create backup rates" message is queue and then processed, asynchronously creating `tax_calculation` entries and `tax_calculation_rate` entries are created.
7. From the same `Stores > Configuration > Sales > Tax` page, once `tax_calculation` entries and `tax_calculation_rate` entries have been created, deactivate "Backup rates" and save configuration.
8. Observe that "Delete backup rates message is queued and processed. TJ `tax_calculation` entries and related `tax_calculation_rate` entries are asynchronously deleted.

#### Versions
